cowbird/app.py

```diff
@@ -22,6 +23,13 @@ def get_app(global_config=None, **settings):
     global_config.update(settings)
     config = get_app_config(global_config)
 
+    # TODO: could be nice to add yaml schema validation
```

Comment:
I'll look into it to add schema validation. I hesitated since the PR was getting bigger.

For the `validate_sync_services_config`, I still think the check is useful here.  It is useful to avoid cases like these :
```
sync_permissions:
  user_workspace:
    services:
      Geoserver:
        workspace:
          - name: api
            type: service
          - ...
      Thredds:
         workspace:
           - name: catalog
             type: service
           - ...
    permissions_mapping:
      - workspace:
        - read
        workspace...?
```
In summary, I want to avoid 2 resources from different services that would use the same key, since these have to be distinguishable in the `permissions_mapping` afterwards.

Reply:
I added some yaml schema validation. I tried the [`schema` package](https://github.com/keleshev/schema) instead of `jsonschema`. Seems to work well so far, and easy to use.

Reply:
The problem in `validate_sync_services_config` is that this mapping: 
```yaml
    permissions_mapping:
      - workspace:
        - read
        workspace: ...?
```
... will never actually happen. 

Give it a try with some dummy YAML. PyYAML simply overrides previous entries by following ones. 
The function will never see a duplicate (since its impossible), and worst will not raise any error giving the false impression everything is good while there is a duplicate in reality. That will cause more confusion.

Reply:
Maybe the example I gave wasn't the best. I added a test case to give a better example of what I meant in this [commit](https://github.com/Ouranosinc/cowbird/pull/21/commits/2268366313da1072b56caeb858526f68245243eb).

In this case, the YAML would not override anything, but there would still be a duplicate key.

Reply:
That use case is good. It's indeed easier to resolve with the config validation function.